packages/react/src/contents/hooks/useCommercePages.ts

```diff
@@ -1,91 +1,87 @@
-  pageIndex?: number,
-  pageSize?: number,
-  strategy = 'default',
+const useCommercePages = <T = ComponentType[]>(
+  options: UseCommercePageOptions,
 ) => {
-  const queryCommerce = {
-    type: params.type,
-    id: params?.id,
-    gender: params?.gender,
-    brand: params?.brand,
-    category: params?.category,
-    priceType: params?.priceType,
-    sku: params?.sku,
-    pageSize,
-    pageIndex,
-  };
-  const queryContent = {
-    codes: slug,
-    contentTypeCode: 'commerce_pages',
-    pageIndex,
-    pageSize,
-  };
+  const {
+    enableAutoFetch = true,
+    fetchConfig,
+    strategy = Strategy.Default,
+    ...fetchQuery
+  } = options;
+
+  const query = useMemo(
+    () => ({
+      contentTypeCode: ContentTypeCode.CommercePages,
+      ...fetchQuery,
+    }),
+    [fetchQuery],
+  );
   const error = useSelector((state: StoreState) =>
-    getContentError(state, queryContent),
+    getContentError(state, query),
   );
   const isLoading = useSelector((state: StoreState) =>
-    isContentLoading(state, queryContent),
+    isContentLoading(state, query),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isContentFetched(state, query),
   );
-  const commercePage = useSelector((state: StoreState) =>
-    getContents(state, queryContent),
+  const entries = useSelector((state: StoreState) =>
+    getContents<T>(state, query),
+  );
+  const contentGroup = useSelector((state: StoreState) =>
+    getContentByQuery(state, query),
+  );
+  const fetchCommercePagesAction = useAction(fetchCommercePages);
+
+  const fetch = useCallback(
+    () => fetchCommercePagesAction(fetchQuery, strategy, fetchConfig),
+    [fetchCommercePagesAction, fetchQuery, strategy, fetchConfig],
+  );
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex: contentGroup?.number,
+      totalPages: contentGroup?.totalPages,
+      totalItems: contentGroup?.totalItems,
+    }),
+    [contentGroup],
   );
-  const action = useAction(fetchCommercePagesAction);
-  const fetchCommercePages = () => action(queryCommerce, slug, strategy);
-  const resetContents = useAction(resetContentsAction);
 
   useEffect(() => {
-    !commercePage && fetchCommercePages();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [commercePage, slug]);
+    if (!isLoading && !error && !isFetched && enableAutoFetch) {
+      fetch();
+    }
+  }, [enableAutoFetch, error, fetch, isFetched, isLoading]);
 
   return {
-    /**
-     * Get the result for a specific Commerce Page.
-     */
-    commercePage,
-    /**
-     * Reset contents.
-     */
-    resetContents,
-    /**
-     * Loading state for a specific Commerce Page.
-     */
     isLoading,
-    /**
-     * Error state for a specific Commerce Page.
-     */
+    isFetched,
     error,
-    /**
-     * Fetch Commerce Page content for a specific slug.
-     */
-    fetchCommercePages,
+    data: {
+      entries,
+      pagination,
+    },
+    actions: {
+      refetch: fetch,
```

Comment:
I think this should be called `fetch` because a `refetch` would only be ok if it was guaranteed that the hook would always make a fetch, which would not be the case if `enableAutoFetch` is false.